internal/services/mssql/mssql_database_resource.go

```diff
@@ -79,39 +87,39 @@ func resourceMsSqlDatabaseImporter(ctx context.Context, d *pluginsdk.ResourceDat
 	replicationLinksClient := meta.(*clients.Client).MSSQL.ReplicationLinksClient
 	resourcesClient := meta.(*clients.Client).Resource.ResourcesClient
 
-	id, err := parse.DatabaseID(d.Id())
+	id, err := commonids.ParseDatabaseID(d.Id())
 	if err != nil {
 		return nil, err
 	}
 
-	partnerDatabases, err := helper.FindDatabaseReplicationPartners(ctx, client, replicationLinksClient, resourcesClient, *id, []sql.ReplicationRole{sql.ReplicationRolePrimary})
+	partnerDatabases, err := helper.FindDatabaseReplicationPartners(ctx, client, replicationLinksClient, resourcesClient, pointer.From(id), []replicationlinks.ReplicationRole{replicationlinks.ReplicationRolePrimary})
 	if err != nil {
 		return nil, err
 	}
 
 	if len(partnerDatabases) > 0 {
 		partnerDatabase := partnerDatabases[0]
 
-		partnerDatabaseId, err := parse.DatabaseID(*partnerDatabase.ID)
+		partnerDatabaseId, err := parse.DatabaseID(*partnerDatabase.Id)
```

Comment:
can we update this to use the `commonids` parser:

```suggestion
		partnerDatabaseId, err := commonids.ParseDatabaseID(*partnerDatabase.Id)
```

Reply:
Fixed.